bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala

```diff
@@ -154,12 +154,13 @@ class BspProjectResolver extends ExternalSystemProjectResolver[BspExecutionSetti
 
   // special handling for sbt projects: run bloopInstall first
   // TODO support other bloop-enabled build tools as well
-  private def preImport(executionSettings: BspExecutionSettings, workspace: File)(implicit reporter: BuildReporter) = {
+  private def preImport(executionSettings: BspExecutionSettings, workspace: File)(implicit reporter: BuildReporter): Try[BuildMessages] = {
     import BspProjectSettings._
 
     val emptySuccess = Success(BuildMessages.empty.status(BuildMessages.OK))
+    def isSbtProject = new File(workspace, "build.sbt").exists()
 
-    if (executionSettings.runPreImportTask) {
+    if (executionSettings.runPreImportTask && isSbtProject) {
```

Comment:
I think logically this belongs more under the if in the `AutoPreImport` case

Reply:
what about `BloopSbtPreImport`? Is it safe to omit it there?

Reply:
That option *should* only be set for sbt projects so it should be safe to omit, and indeed it should be an error if it is set for non-sbt projects.